src/lib/userStorage/UserStorageClass.js

```diff
@@ -735,60 +842,102 @@ export class UserStorage {
    * @returns {Promise} Promise with StandardFeed object,
    *  with props { id, date, type, data: { amount, message, endpoint: { address, displayName, avatar, withdrawStatus }}}
    */
-  formatEvent = memoize(
-    // eslint-disable-next-line require-await
-    (event: FeedEvent) => {
-      try {
-        logger.debug('formatEvent: incoming event', event.id, { event })
-        const { data, type, date, id, status, createdDate, animationExecuted, action } = event
-        const { sender, preReasonText, reason, code: withdrawCode, subtitle, readMore, smallReadMore } = data
-
-        const { address, initiator, initiatorType, value, displayName, message, avatar } = this._extractData(event)
-
-        // displayType is used by FeedItem and ModalItem to decide on colors/icons etc of tx feed card
-        const displayType = this._extractDisplayType(event)
-        logger.debug('formatEvent: initiator data', event.id, {
-          initiatorType,
-          initiator,
-          address,
-        })
+  // eslint-disable-next-line require-await
+  async formatEvent(event: FeedEvent) {
+    return this._cacheFormattedEvent(event, async () => {
+      logger.debug('formatEvent: incoming event', event.id, { event })
+
+      const { feedStorage } = this
+      const { data, type } = event
+      const { counterPartyFullName, counterPartySmallAvatar } = data
+
+      const counterPartyEvents = [
+        FeedItemType.EVENT_TYPE_SENDDIRECT,
+        FeedItemType.EVENT_TYPE_SEND,
+        FeedItemType.EVENT_TYPE_WITHDRAW,
+        FeedItemType.EVENT_TYPE_RECEIVE,
+      ]
+
+      if (counterPartyEvents.includes(type) && (!counterPartyFullName || !counterPartySmallAvatar)) {
+        const counterPartyData = await feedStorage.getCounterParty(event)
```

Comment:
this is the place where we want to check if counter party was updated.
you are getting the counter party from the cache but not trying to update the cache if it empty and 24hours passed since last check

Reply:
getCounterParty checks for this and updates the cache

Reply:
ok you are right, i missed that